app/code/community/Webbhuset/Features/Model/Sitemap.php

```diff
@@ -0,0 +1,263 @@
+{
+    
+    /**
+    * Sitemap file number
+    *
+    * @var int
+    */
+    protected $_fileNumber = 0;
+
+    /**
+    * Sitemap split limit
+    *
+    * @var int
+    */
+    protected $_limit = null;
+
+    /**
+    * Counter towards split limit
+    *
+    * @var int
+    */
+    protected $_index = 0;
+
+    /**
+     * Generate Sitemap Index XML file
+     */
+    protected function _createSitemapIndexFile()
+    {
+        $io = new Varien_Io_File();
+        $io->setAllowCreateFolders(true);
+        $io->open(array('path' => $this->getPath()));
+        $storeId = $this->getStoreId();
+        $date    = Mage::getSingleton('core/date')->gmtDate('Y-m-d H:i:s');
+        $baseUrl = Mage::app()->getStore($storeId)->getBaseUrl(Mage_Core_Model_Store::URL_TYPE_LINK);
+        $sitemapFileName = $this->getSitemapFilename();
+
+        $io->streamOpen($sitemapFileName);
+        $io->streamWrite('<?xml version="1.0" encoding="UTF-8"?>' . "\n");
+        $io->streamWrite('<siteMapindex xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">');
+
+        for ($i = 1; $i < $this->_fileNumber; $i++) {
+            $fileName = $this->_getFilename($i);
+            $xml = sprintf(
+                '<sitemap><loc>%s</loc><lastmod>%s</lastmod></sitemap>',
+                $this->_getSitemapUrl($fileName),
+                $date
+            );
+            $io->streamWrite($xml);
+        }
+        $io->streamWrite('</siteMapindex>');
+        $io->streamClose();
+    }
+    
+    /**
+    * Return url for Sitemap file
+    *
+    * @param String $fileName
+    * @return String
+    */
+    protected function _getSitemapUrl($fileName)
+    {
+
+        $fileName = trim($this->getSitemapPath() . $fileName, '/');
+        $url = Mage::helper('sitemap')->escapeHtml(
+            Mage::app()->getStore($this->getStoreId())->getBaseUrl(Mage_Core_Model_Store::URL_TYPE_WEB) . $fileName
+        );
+
+        return $url;
+    }
+
+    /**
+     * Get fileName for subSitemap XML file
+     * 
+     * @param int $number
+     * @return String
+     */
+    protected function _getFilename($number = null)
+    {
+        if ($number === null) {
+            $number = $this->_fileNumber;
+        }
+        $sitemapFileName = $this->getSitemapFilename();
+        $fileName = sprintf(
+            '%s-%02d.xml',
+            basename($sitemapFileName, '.xml'),
+            $number
+        );
+
+        return $fileName;
+    }
+
+    /**
+     * check if new SubSitemap XML file is needed.
+     *
+     * @param Varien_Io_File $io
+     * @return Varien_Io_File
+     */
+    protected function _checkFile($io) 
+    {
+        if ($this->_limit == null) {
+            $this->_limit = Mage::getStoreConfig('sitemap/generate/split_limit');
+        }
+        if ($this->_index == $this->_limit) {
+
+            $this->_closeFile($io);
+            $io = $this->_newFile();
+            $this->_index = 0;
+            $this->_index += 1;
```

Comment:
Detta kan skrivas lite snyggare. 

Eftersom du alltid inkrementerar  `$this->_index` efter loopen kan du ta bort denna rad och return-statement i if-satsen.

```php
... code ...
if ($this->_index == $this->_limit) {
    $this->_closeFile($io);
    $io = $this->_newFile();
    $this->_index = 0;
}
$this->_index += 1;

return $io;
```